scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import (absolute_import, division, print_function)
+
+from six import iteritems
+
+import mantid.simpleapi as mantid
+
+
+class FFTWrapper(object):
+
+    """
+    A class to wrap the different parts
+    of the FFT and its preprocessing.
+    This keeps the main FFT class simple.
+    """
+
+    def __init__(self, FFT):
+        self.name = "FFT"
+        self.model = FFT
+
+    def cancel(self):
+        self.model.cancel()
+
+    def loadData(self, inputs):
+        """
+        store the data in the wrapper for later
+        """
+        if "phaseTable" in inputs:
+            self.phaseTable = inputs["phaseTable"]
+        else:
+            self.phaseTable = None
+        if "preRe" in inputs:
+            self.preRe = inputs["preRe"]
+        else:
+            self.preRe = None
+        if "preIm" in inputs:
+            self.preIm = inputs["preIm"]
+        else:
+            self.preIm = None
+        if "FFT" in inputs:
+            self.FFT = inputs["FFT"]
+        else:
+            self.FFT = None
+        self.model.setRun(inputs["Run"])
+
+    def execute(self):
+        """
+        runs the relevant parts of the FFT and the preprocessing
+        """
+        if self.phaseTable is not None:
+            if self.phaseTable["newTable"]:
+                self.model.makePhaseQuadTable(self.phaseTable)
+            self.model.PhaseQuad()
+
+        if self.preRe is not None:
+            self.model.preAlg(self.preRe)
+
+        if self.preIm is not None:
+            self.model.preAlg(self.preIm)
+
+        if self.FFT is not None:
+            self.model.FFTAlg(self.FFT)
+
+    def output(self):
+        return
+
+
+class FFTModel(object):
+
+    """
+    A simple class which executes
+    the relevant algorithms for
+    the analysis.
+    """
+
+    def __init__(self):
+        self.name = "FFT"
+        self.alg = None
+
+    def cancel(self):
+        if self.alg is not None:
+            self.alg.cancel()
+
+    def setRun(self, run):
+        self.runName = run
+
+    def preAlg(self, preInputs):
```

Comment:
This name confused me... is it implying that this algorithm must be performed before FFT?

Reply:
yes